apps/core/scenes/search/Search.js

```diff
@@ -80,36 +87,69 @@ class Search extends React.Component<Props> {
     this.props.setModalType(MODAL_TYPE.TRIP_TYPE);
   };
 
+  handleTripTypeSelect = (type: string) => {
+    switch (type) {
+      case 'return':
+      case 'oneWay':
+        this.props.setTripType(type);
+        break;
+      default:
+        break;
+    }
+  };
+
   handlePassengersPress = () => {
     this.props.setModalType(MODAL_TYPE.PASSENGERS);
   };
 
   render() {
-    const { tripType, adults, infants, bags } = this.props;
+    const { tripType, adults, infants, bags, layout } = this.props;
+    const desktopLayout = layout >= LAYOUT.desktop;
+
     return (
       <>
         <View style={styles.container}>
-          <View style={styles.form}>
+          <View style={[styles.form, desktopLayout && styles.formDesktop]}>
             <Illustration name="Boarding" style={styles.boardingIllustration} />
             <View style={styles.top}>
-              <TripTypeButton
-                onPress={this.handleTripTypePress}
-                icon={TRIP_TYPE[tripType].icon}
-                label={TRIP_TYPE[tripType].label}
-              />
+              {Platform.OS === 'web' ? (
+                <TripTypeSwitch
+                  style={desktopLayout && styles.flexTripTypeSwitch}
+                  optionsData={TRIP_TYPE}
+                  selectedType={tripType}
+                  onSelect={this.handleTripTypeSelect}
+                />
+              ) : (
+                <TripTypeButton
+                  onPress={this.handleTripTypePress}
+                  icon={TRIP_TYPE[tripType].icon}
+                  label={TRIP_TYPE[tripType].label}
+                />
+              )}
```

Comment:
Maybe we should consider extracting this to a separate component. This component is already too large, and we should not stick more logic in here

Reply:
Good point, definitely, I'll do it.

Reply:
`TripType` and `PassengersButton` section extracted into separate component. So search form is way much lighter now.